ssl/quic/quic_demux.c

```diff
@@ -442,6 +442,7 @@ int ossl_quic_demux_inject(QUIC_DEMUX *demux,
 
     /* Move from free list to pending list. */
     ossl_list_urxe_remove(&demux->urx_free, urxe);
+    urxe->datagram_id = demux->next_datagram_id++;
     ossl_list_urxe_insert_tail(&demux->urx_pending, urxe);
     urxe->demux_state = URXE_DEMUX_STATE_PENDING;
 
```

Comment:
```c
QUIC_URXE *urxe;
for (i = 0; i < num_urxe; i++) {
    urxe = get_urxe(i); // Hypothetical function to get urxe
    if (urxe == NULL) {
        BIO_printf(bio_err, "Failed to get urxe at index %d\n", i);
        goto end;
    }
    if (!ossl_list_urxe_insert_tail(&demux->urx_pending, urxe)) {
        BIO_printf(bio_err, "Failed to insert urxe at index %d\n", i);
        goto end;
    }
}
```

Reply:
sorry for not providing more details here in PR. it fixes issue reported by bernd in related #26277. The stack traces he gathered indicate code trips sanitizer alarm when logs packet which is received by t-server via `ossl_quic_demux_inject()`.